File: pascal.py
def pascal(r, c):
    """ Calculate a given row/column value in a Pascal Triangle """
    if r > c:
        return None
    if r == 0:
        return 1
    if r == c:
        return 1
    else:
        return pascal(max(r-1,0), max(c-1,0)) + pascal(max(r,0), max(c-1,0))

if __name__ == '__main__':
    print pascal(2,2)

File: pascal.scala
object Pascal_Triangle {
  // Calculate a given row/column value in a Pascal Triangle
  def pascal(r: Int, c: Int): Int = (r, c) match {
    case (r, c) if r > c => 0 
    case (0, _)  => 1 
    case (r, c) if r == c => 1
    case _ => pascal(r-1, c-1) + pascal(r, c-1)
  }

  def main(args: Array[String]) {
    println(pascal(3,4))
  }
}
